cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py

```diff
@@ -88,17 +89,16 @@ def discard_index_sequence(sample_id_with_index: str) -> str:
 def get_bcl2fastq_stats_paths(demultiplex_result_directory: Path) -> List[Path]:
     """Find metrics files in Bcl2fastq demultiplex result directory."""
     stats_json_paths = []
-    pattern = re.compile(r"l\d+t\d+")
-
-    for subdir in os.listdir(demultiplex_result_directory):
-        if pattern.match(subdir):
-            stats_json_path = (
-                demultiplex_result_directory
-                / subdir
-                / BCL2FASTQ_METRICS_DIRECTORY_NAME
-                / BCL2FASTQ_METRICS_FILE_NAME
-            )
-            if stats_json_path.is_file():
+
+    for root, dirs, files in os.walk(demultiplex_result_directory):
```

Comment:
```suggestion
    for root, dirs, files in os.walk(demultiplex_result_directory):
```
We have a function for this in /cg/utils/files/ 
that one is tested and we should use the implementation of it here. 
`get_file_in_directory()`

Reply:
Good call! Unfortunately we want all examples of the Stats.json file and `get_file_in_directory` seems to only return the first match